addons/batikvis/src/main/java/de/lmu/ifi/dbs/elki/visualization/VisualizationTask.java

```diff
@@ -248,7 +248,7 @@ public int compareTo(VisualizationTask other) {
     // sort by name otherwise.
     String name1 = this.getMenuName();
     String name2 = other.getMenuName();
-    if(name1 != null && name2 != null && name1 != name2) {
+    if(name1 != null && name2 != null && !name1.equals(name2)) {
       return name1.compareTo(name2);
```

Comment:
Note that the actual comparison happens here. The identity check is a short-circuit, probably redundant.